app/models/assignment.rb

```diff
@@ -576,12 +580,15 @@ def summary_json(user)
                                .map { |s| [s['users.user_name'], s['users.first_name'], s['users.last_name']] }
       end
 
+      tag_info = tag_data.fetch(g.id, {}
```

Comment:
The logic in this part isn't right. You have a `{}.map ...` which I don't think you want.

The `fetch` isn't a bad idea, but you want the default value to be `[]`, and the `map` to go on the outside. After you do that you'll find that you can just do `tags: tag_info` below.